### ISSUE TITLE
App is Non-Functional Without KVKit

### ISSUE THREAD

**Issue by noodletoad** [fancherblack/rqst#1]
Without KVKit, the request process will not work with recent versions of Splunk due to the removal of HTML dashboards. 

KVKit does not seem to be available, though :-(

**Comment by jaybenfield** [comment]
Hey @noodletoad, Splunk's deprecation of HTML dashboards does break the app's request form functionality. The request mechanism was the most labor-intensive aspect of the app's development, _thus please insert weeping and gnashing of teeth here_. 

I am open to making the kvkit codebase (Node.js/Express) available to the Splunk community. However, getting the application ready for use would require some effort. Beyond a general refresh and tidying, the biggest lift would be the decoupling/whacking of the license verification mechanism.

If I did make kvkit available, would you be interested in contributing?

**Comment by noodletoad** [comment]
Yes I would! Being able to quickly create a UI form to edit KV Store data is something that would be super useful - I’m sure other’s feel the same way since most of us have been defaulting to KV Store lookups vs CSV for years now. Sure, there’s the Lookup Editor App but that’s clunky and dangerous for end-user use. 

For context, I’ve been doing Splunk Professional Services for 10 years now, mostly in Enterprise Security. I have a software development background too so I know the pain of the license code process… ick.

One more thing I noticed - the Github repo is missing user_workspace.js, so even enabling legacy jQuery libraries doesn’t make the request functionality work :-( 

-------------------------
Dennis Morton
Principal Consultant
Arcus Data
Mobile: 214-289-3675

**Comment by jaybenfield** [comment]
That's great news. Thanks, @noodletoad! 

In early 2020, I was on a Splunk PS engagement with a large customer developing an app with custom KV Store functionality. Over the course of the engagement, I spent hours discussing the need for better KV Store capabilities with another Splunk consultant. The outcome of all that _imagineering_ and Splunky philosophizing was a decision to take a run at the problem. Much of _Bring Your Child to Work Year_ was spent developing kvkit with my son Josh, a very talented developer. We proved the concept and ultimately piloted the app with the aforementioned customer, and in several other environments. 

Hindsight being 20/20, we _should have_ open-sourced the app once we'd successfully piloted it. Instead, we attempted to monetize and spent a silly amount of time and energy building our own license server setup, coming up with a code obfuscation strategy, etc. Bleh. Gross. Ultimately, it was pushed to the back-burner by our day jobs and other projects with dollars attached. 

As for me, I've been in Splunk's orbit since 2010-ish. I've been a customer, a Splunk employee, a Splunk ISV, and a consultant with a Splunk partner. Today, I am a customer once again - though I still do some extracurricular Splunkery in my free time. 

I will resurrect kvkit, get reacquainted, take a pass at some of the known issues, and drop it in a public repo. With the holiday and some travel next week, we're probably looking at a week or two before I can swing this, but I'll definitely let you know once it's online. 

Thanks again!

**Comment by jaybenfield** [comment]
I just realized I didn't reply to your comment about user_workspace.js. I'll also see if I can track that down!

**Comment by jaybenfield** [comment]
Hey, @noodletoad. The kvkit application is now freely available: [https://github.com/fancherblack/kvkit](https://github.com/fancherblack/kvkit)
